### DISCUSSION TITLE
Use of spaces in QQL

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#260] [Q&A]
Can we query using spaces in qql in shell or TimeBase admin UI? 
Does Community or enterprise edition supports such querying?

**Comment by alex-karpovich** [comment]
@ankitsoin  TimeBase admin UI support spaces: 

![image](https://github.com/finos/TimeBase-CE/assets/18558216/3be327f4-48b2-424e-8d12-fcaef1ab08f5)

![image](https://github.com/finos/TimeBase-CE/assets/18558216/bc737960-820c-439f-ba2f-371c06c9ce35)

**Reply by ankitsoin** [reply]
Hi @alex-karpovich,

Display spaces option above just displays spaces.
We need to know can we query using some specific space.
Example: let say my space is 20231101 and can I query like this select * from stream where space=='20231101'? It isnt working

Or any other way to query space?

I can see space() function coming in TB UI editor as suggestion but not sure how to use that function.

We tried right clicking  on the name of the space and the selecting 'query' from options list but that also didnt work.

**Reply by alex-karpovich** [reply]
what language you are using? Java or Python?

**Reply by ankitsoin** [reply]
Actually ask is how to use spaces in QQL, not sure how to use "space()" function in timebase admin

**Reply by ShivaniSK** [reply]
Hello @alex-karpovich as Ankit mentioned, we are looking for a way to query inside a particular space using Timebase Web Admin.
We are able to see the list of spaces in the Web Admin, however cannot find a way to query inside a specific space.

**Reply by alex-karpovich** [reply]
@ShivaniSK via Query Editor it's not possible now. We will extends QQL grammar to support `spaces`
